lib/crystalball/predictor.rb

```diff
@@ -32,7 +32,12 @@ def prediction
     alias cases prediction
 
     def diff
-      repo.diff(from, to)
+      @diff ||= repo.diff(merge_base, to)
+    end
+
+    def merge_base
+      to_or_head = to ? to : 'HEAD'
+      `git --git-dir=#{repo.repo.path} --work-tree=#{repo.dir.path} merge-base #{from} #{to_or_head} 2>&1`.strip
```

Comment:
![No!](https://media3.giphy.com/media/d1E1msx7Yw5Ne1Fe/giphy-downsized.gif)
Please do it on proper level. There should be a method for repo which uses ruby-git lib command stuff etc. merge-base is useful general purpose command which we should try to submit to ruby-git library

Reply:
Someone tried that [in 2012](https://github.com/ruby-git/ruby-git/pull/43). It's hard to get anything merged there. But I agree that having it on a higher level of abstraction would benefit our own code base, making it easier to reason about, test, etc. Maybe that PR I linked is a good starting point.